### ISSUE TITLE
Not clear how to run Powersync-only using `docker compose up`

### ISSUE THREAD

**Issue by guillempuche** [powersync-ja/self-host-demo#20]
Ideal case: start Powersync services in one single command line `docker compose up -d`. Pointed also here https://discord.com/channels/1138230179878154300/1194710422960472175/1263540402968068147.

Right now, it's split in different places, moreover, it's not clear.
- Which Docker service to start first
- Just run Powersync first, then frontend and backend. Where do I find the Powersync services. Example: terminal 1 `cd ¿? && docker compose up -d` + terminal 2 `cd ¿? && docker compose up -d`?. People can confuse if https://github.com/powersync-ja/self-host-demo/tree/main/services and https://github.com/powersync-ja/self-host-demo/blob/main/config/powersync.yaml have Powersync-only services or also demo-related.

The docs web could also explain running pure Powersync (no frontend & backend demo), like in the first sections of https://supabase.com/docs/guides/cli/local-development until here https://supabase.com/docs/guides/cli/local-development#start-supabase-services.

**Comment by stevensJourney** [comment]
Thanks for the feedback. We are busy compiling docs for this.

Currently the demo repo's readmes do have single commands for starting the demos, but I'm assuming you're referring to it not being clear how to construct a Docker compose project from scratch which can be started with a single command. 

In short there are the following requirements:

- A MongoDB replica set service. This should be configured as a replica set and be started before the PowerSync service is started. Currently we use a `depends_on` in the `powersync` service to achieve this.
- A Postgres service. There aren't any strict requirements for this to be started before the PowerSync service is started, but it does cause less error logs if started before. 
- An authentication mechanism. This is where there are a few options. Essentially the PowerSync service needs access to the public key in order to verify JWTs. In the Demos we have JWKS endpoints exposed by the backends which the PowerSync service queries for the public key. The public key could also be specified in the `powersync.yaml` config file.

**Comment by kobiebotha** [comment]
Hi @guillempuche , we put together some docs here https://docs.powersync.com/self-hosting/local-development

Could you take a look and let us know if you have any feedback?

**Comment by guillempuche** [comment]
@kobiebotha,x I'm running the default Supabase docker. When running PowerSync docker, this error happens:
```
[+] Running 6/7
 ✔ Network powersync_default            Created                                                                               0.0s 
 ✔ Volume "powersync_pg_data"           Created                                                                               0.0s 
 ✔ Volume "powersync_mongo_storage"     Created                                                                               0.0s 
 ⠴ Container powersync-postgres-1       Starting                                                                              0.6s 
 ✔ Container powersync-mongo-1          Started                                                                               0.6s 
 ✔ Container powersync-mongo-rs-init-1  Created                                                                               0.0s 
 ✔ Container powersync-powersync-1      Created                                                                               0.0s 
Error response from daemon: driver failed programming external connectivity on endpoint powersync-postgres-1 (e341870be68cb666141c8bd66ec229c541bf693952ed111d853d1368c0059cac): Bind for 0.0.0.0:5432 failed: port is already allocated
```

Tomorrow, I'll post the solution.

---

My `config.yml`:

```yml
replication:
  connections:
    - type: postgresql
      uri: postgresql://postgres:postgres@postgres:5432/postgres

      # SSL settings
      sslmode: disable # 'verify-full' (default) or 'verify-ca' or 'disable'

# Connection settings for sync bucket storage
storage:
  type: mongodb
  uri: mongodb://mongo:27017/powersync_demo

# The port which the PowerSync API server will listen on
port: 8080

sync_rules:
  # TODO use specific sync rules here
  content: |
    bucket_definitions:
      global:
        data:
          - SELECT * FROM creators
          - SELECT * FROM profiles

client_auth:
  # Enable this if using Supabase Auth
  supabase: true

  # JWKS URIs can be specified here if you were using a custom setup
  # jwks_uri: [TODO]  # This is not needed when supabase: true is set

  # JWKS audience
  audience: ["powersync-dev", "powersync"]

# Settings for telemetry reporting
# See https://docs.powersync.com/self-hosting/telemetry
telemetry:
  # Opt out of reporting anonymized usage metrics to PowerSync telemetry service
  disable_telemetry_sharing: false
```

**Comment by stevensJourney** [comment]
This part of the error seems to be the issue: `Bind for 0.0.0.0:5432 failed: port is already allocated`. In this case you might have another container already binding to port `5432`. Stopping the other container or changing the port should solve the issue. 

If using Supabase's local development for Postgres, you don't need to provide a Postgres service in your Docker compose file.

**Comment by guillempuche** [comment]
Deleting `postgres` in Powersync Docker compose, Powersync docker container starts.

- Can I open a Powersync UI like Supabase `http://localhost:8000`, or is all via cmd? 
- For the connections URI in `config.yaml`, do I need to change something if I use the default `.env`  of Supabase https://github.com/supabase/supabase/blob/master/docker/.env.example?

**Comment by stevensJourney** [comment]
That's good to hear :)

- Unfortunately we do not host any frontend UI from the PowerSync service. All the configuration is achieved by editing the config YAML file.
- It has been a while since I tried the standard Supabase project config. The last time I tried I noticed that the default Postgres service was exposed on a high port number somewhere in the 52000 range. We currently don't support connections on ports in this range. Our demo overrides this port in the Supabase config to 5433.
